### DISCUSSION TITLE
ImageMagick Example Usage

### DISCUSSION THREAD

**Discussion by urban-warrior** [ImageMagick/ImageMagick#5178] [General]
We modernized the legacy ImageMagick example usage documentation to version 7 tools and options.  It previously reflected legacy version 6 tools and options.  We also modernized the CSS for an improved user experience on the desktop and mobile devices.  You can now cut-n-paste example commands directly from the web page to your command-line to reproduce the usage results discussed on the web pages.  Windows users, replace the backslash `\` with `^`.

You can find the ImageMagick version 7 example usage @ https://imagemagick.org/Usage/.

Thanks to @fmw42 for reviewing the web pages for tool and option correctness.

**Comment by snibgo** [comment]
Excellent. I expect that was a major effort, and the results look good.

Each page has a trailer. For example, https://imagemagick.org/Usage/layers/ has a trailer that includes:
```
Examples Generated with: IM v6.6.5-3 2010-10-23
```
... where the text "IM v6.6.5-3 2010-10-23" is an image. I suppose the text is incorrect.

**Reply by urban-warrior** [reply]
We'll blank them for now and update the version when we rerun Anthony's scripts.  Recall, the image may be cached so wipe your image cache if you want to see the correction.

**Comment by GeeMack** [comment]
Thanks for all the time and effort. I've discovered a minor error on this page, the "Anatomy of the Command-line"...

[https://imagemagick.org/script/command-line-processing.php](url)

An example command I contributed is near the top of the page. Near the end of the command there's a distort operation that is split with a continued-line backslash"\\"...

    ... -distort SRT "0.96 \
        %[fx:rand()*360]" ... 

Although this should work as a *nix command, if the backslashes "\\" are changed to carets "^" to make it into a Windows command, it will fail...

    ... -distort SRT "0.96 ^
        %[fx:rand()*360]" ... 

For Windows a continued-line caret "^" can't be inside quotes like that. If someone can edit that page so it doesn't split that `-distort` operation, that would be great. Thanks!

**Reply by fmw42** [reply]
Good catch.

**Comment by Ryan-DowlingSoka** [comment]
The new CLI examples are great.

I'd love if a Windows/Linux toggle could be implemented on the code boxes. If there is interest in the idea I would be happy to provide the html and JavaScript to do so. My intention would be to do the transformation automatically through regex from the Linux version, to avoid needing to maintain two versions.

It could also be for the entire page instead of per box if that is preferred.

**Reply by urban-warrior** [reply]
Post a pull request to https://github.com/ImageMagick/Usage with your commits and we will review and merge.
